esmvalcore/_config.py

```diff
@@ -44,18 +45,28 @@ def read_config_user_file(config_file, folder_name, options=None):
     with open(config_file, 'r') as file:
         cfg = yaml.safe_load(file)
 
+    # DEPRECATED: remove in v2.3
+    for setting in ('write_plots', 'write_netcdf', 'output_file_type'):
+        if setting in cfg:
+            msg = (
+                f"Using '{setting}' in {config_file} is deprecated and will "
+                "be removed in ESMValCore version 2.3. For diagnostics "
+                "that support this setting, it should be set in the "
+                "diagnostic script section of the recipe instead. "
+                f"Remove the setting from {config_file} to get rid of this "
+                "warning message.")
+            print(f"Warning: {msg}")
+            warnings.warn(DeprecationWarning(msg))
+
     if options is None:
         options = dict()
     for key, value in options.items():
         cfg[key] = value
 
     # set defaults
     defaults = {
-        'write_plots': True,
```

Comment:
I think we should keep the defaults for now to ensure no diagnostic code breaks until 2.3

Reply:
Removing the defaults would not break any diagnostic code, but it does break some recipes if you do not have the parameters in your config-user file. The downside of not removing them is that there might be not enough incentive to update any recipe. Maybe you're right and this is too strong though. Shall I add them back in?

Reply:
yeah I reckon the defaults should be kept, I can see a reason why keeping them even after 2.3 since we may want to revisit some in the future

Reply:
Restored the defaults!